pkg/reconciler/serverlessservice/resources/services.go

```diff
@@ -108,6 +109,13 @@ func MakePrivateService(sks *v1alpha1.ServerlessService, selector map[string]str
 				// This one is matching the public one, since this is the
 				// port queue-proxy listens on.
 				TargetPort: targetPort(sks),
+			}, {
+				// Expose the QueueAdminPort in order to allow PreStop hooks to be called.
+				// This is important for graceful shutdown of user-container.
```

Comment:
I think we need a much more detailed comment here that highlights *why* we need this.  See: https://github.com/knative/serving/pull/5544/files#diff-60ed5784d23125a1d08e8de7d0458e6cR520-R524

Reply:
Agreed, done.